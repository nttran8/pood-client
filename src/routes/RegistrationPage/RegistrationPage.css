.RegistrationPage {
	padding: 25px 25px 20px 25px;
	display: flex;
	flex-direction: column;
	justify-content: space-evenly;
}

.sec {
	display: flex;
	flex-direction: column;
	align-items: center;
}

.Page__left {
	display: none;
}

@media only screen and (min-width: 576px) {
	.RegistrationPage {
		padding: 60px 10px 20px 10px;
		flex-direction: row;
		justify-content: space-around;
	}

	.Page__left {
		display: block;
	}

	.Page__left > p {
		font-size: 23px;
		width: 40vw;
		text-align: left;
	}

	.vision {
		margin-top: 50px;
		width: 40vw;
	}

	.Page__right {
		width: 40vw;
		height: 100vh;
		gap: 7rem;
	}
}

@media only screen and (min-width: 768px) {
	.RegistrationPage {
		padding: 50px 20px 20px 20px;
		justify-content: space-evenly;
	}

	.Page__left {
		margin-left: 40px;
	}

	.Page__left > p {
		font-size: 25px;
		font-weight: 600;
		width: 40vw;
	}

	.vision {
		position: relative;
		left: 13%;
		bottom: 18%;
		width: 35vw;
	}

	.Page__right {
		padding: 0 2rem;
		gap: 7rem;
	}
}
