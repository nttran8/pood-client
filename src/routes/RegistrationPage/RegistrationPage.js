// Libraries
import React, { Component } from 'react';

// Component
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm';

// Style
import './RegistrationPage.css';
import registerImg from '../../img/Register.svg';

export default class RegisterPage extends Component {
  static defaultProps = {
    history: {
      push: () => {}
    }
  };

  handleHomeRedirect = () => {
    // Render homepage/login page after user is registered
    const { history } = this.props;
    history.push('/');
  }

  render() {
    return (
      <section className='RegistrationPage'>
        <div className="Page__left sec">
          <p>Just doo it.</p>
          <img className="vision" src={registerImg} alt="character being dram"/>
        </div>
        <div className="Page__right sec">
        <RegistrationForm 
          onRegistration={this.handleHomeRedirect}
        />
        </div>
      </section>
    );
  }
}
