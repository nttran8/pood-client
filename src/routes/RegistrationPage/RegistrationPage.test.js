import React from "react";
import TestRenderer from "react-test-renderer";
import RegistrationPage from "./RegistrationPage";
import { library } from "@fortawesome/fontawesome-svg-core";
import { faUser, faEnvelope, faLock } from "@fortawesome/free-solid-svg-icons";
library.add(faUser, faEnvelope, faLock);

it(`Renders RegistrationPage UI as expected`, () => {
  const tree = TestRenderer.create(<RegistrationPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
