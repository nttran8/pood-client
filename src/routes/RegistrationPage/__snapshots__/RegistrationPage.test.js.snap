// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders RegistrationPage UI as expected 1`] = `
<section
  className="RegistrationPage"
>
  <div
    className="Page__left sec"
  >
    <p>
      Let's doo it.
    </p>
    <img
      alt="character being dram"
      className="vision"
      src="Register.svg"
    />
  </div>
  <div
    className="Page__right sec"
  >
    <div
      className="RegistrationForm__logoBox"
    >
      <img
        alt="logo"
        className="RegistrationForm__logo"
        src="Pood.svg"
      />
    </div>
    <form
      className="RegistrationForm"
      onSubmit={[Function]}
    >
      <div
        className="formBox"
      >
        <h2>
          Create Account
        </h2>
        <div
          role="alert"
        />
        <div
          className="RegistrationForm__credential"
        >
          <div
            className="inputBox"
          >
            <label
              aria-label="username"
              htmlFor="username"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-user fa-w-14 "
                data-icon="user"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </label>
            <input
              className="input username publicPlaceholder"
              id="username"
              placeholder="Username"
              required={true}
            />
          </div>
          <div
            className="inputBox"
          >
            <label
              aria-label="email"
              htmlFor="email"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-envelope fa-w-16 "
                data-icon="envelope"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </label>
            <input
              className="input email publicPlaceholder"
              id="email"
              placeholder="Email"
              required={true}
            />
          </div>
          <div
            className="inputBox"
          >
            <label
              aria-label="password"
              htmlFor="password"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-lock fa-w-14 "
                data-icon="lock"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </label>
            <input
              className="input password publicPlaceholder"
              id="password"
              placeholder="Password"
              required={true}
              type="password"
            />
          </div>
        </div>
        <button
          className="button "
          type="submit"
        >
          Register
        </button>
        <button
          className="button hideButton"
          onClick={[Function]}
          type="button"
        >
          Back
        </button>
      </div>
    </form>
  </div>
</section>
`;
